cmd/nerdctl/create_linux_test.go

```diff
@@ -33,3 +35,79 @@ func TestCreate(t *testing.T) {
 	base.Cmd("start", tID).AssertOK()
 	base.Cmd("logs", tID).AssertOutContains("foo")
 }
+
+func TestCreateWithMACAddress(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	networkBridge := "testNetworkBridge" + tID
+	networkMACvlan := "testNetworkMACvlan" + tID
+	networkIPvlan := "testNetworkIPvlan" + tID
+	base.Cmd("network", "create", networkBridge, "--driver", "bridge").AssertOK()
+	base.Cmd("network", "create", networkMACvlan, "--driver", "macvlan").AssertOK()
+	base.Cmd("network", "create", networkIPvlan, "--driver", "ipvlan").AssertOK()
+	t.Cleanup(func() {
+		base.Cmd("network", "rm", networkBridge).AssertOK()
+		base.Cmd("network", "rm", networkMACvlan).AssertOK()
+		base.Cmd("network", "rm", networkIPvlan).AssertOK()
+	})
+	tests := []struct {
+		Network string
+		WantErr bool
+		Expect  string
+	}{
+		{"host", true, "conflicting options"},
+		{"none", true, "can't open '/sys/class/net/eth0/address'"},
+		{"container:whatever" + tID, true, "conflicting options"},
+		{"bridge", false, ""},
+		{networkBridge, false, ""},
+		{networkMACvlan, false, ""},
+		{networkIPvlan, true, "not support"},
+	}
+	for i, test := range tests {
+		containerName := fmt.Sprintf("%s_%d", tID, i)
+		macAddress, err := nettestutil.GenerateMACAddress()
+		if err != nil {
+			t.Errorf("failed to generate MAC address: %s", err)
+		}
+		if test.Expect == "" && !test.WantErr {
+			test.Expect = macAddress
+		}
+		cmd := base.Cmd("create", "--network", test.Network, "--mac-address", macAddress, "--name", containerName, testutil.CommonImage, "cat", "/sys/class/net/eth0/address")
+		if !test.WantErr {
+			t.Cleanup(func() {
+				base.Cmd("rm", containerName).AssertOK()
+			})
+			cmd.AssertOK()
+			base.Cmd("start", containerName).AssertOK()
+			cmd = base.Cmd("logs", containerName)
+			cmd.AssertOK()
+			cmd.AssertOutContains(test.Expect)
+		} else {
+			if (testutil.GetTarget() == testutil.Docker && test.Network == networkIPvlan) || test.Network == "none" {
+				// 1. unlike nerdctl
+				// when using network ipvlan in Docker
+				// it delays fail on executing start command
+				// 2. start on network none will success in both nerdctl and
+				// Docker, so we need to remove it on finish
+				t.Cleanup(func() {
+					base.Cmd("rm", containerName).AssertOK()
+				})
```

Comment:
77-79, 92-94 Can we merge them?

Reply:
Deleting containers that failed to create has no side effects.

Reply:
LGTM